File: array/TopKFreq/TopK.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number[]}
 */
var topKFrequent = function(nums, k) {
    map = {};
    nums.forEach(function(num){//anamous callback
        if(!map[num]) map[num] = 1;
        else map[num]++;
    });
    var temp = [];
    for (var m in map){ //loop through all keys
        temp.push({key:m,value:map[m]});
        console.log(m);
    }
    temp.sort(function(x,y){
        return y.value-x.value //customer reverse sort
    })
    var result = [];
    for (var i =0;i<k;i++){
        result.push(+temp[i].key);  // key value pairs all K keys for most frequent + for cast to int
    }
    return result;
};

File: array/TopKFreq/TopK.py

class Solution:
    def topKFrequent(self, nums, k):
        record = {}
        for i in nums:
            if i in record:
                record[i] +=1
            else:
                record[i] = 1
        Srecord = sorted(record.items(),key = operator.itemgetter(1),reverse=True)# sort by second in pairs
        result = []
        for key,value in Srecord: #append all to result
            result.append(key)
        return result[:k]
